src/openfluid/ui/waresdev/GitUIProxy.cpp

```diff
@@ -225,9 +231,91 @@ bool GitUIProxy::addSubmodule(const QString& FromUrl, const QString& ToPath, con
 // =====================================================================
 
 
+bool cleanModuleDir(QString RootPath, openfluid::tools::FilesystemPath ModuleSubPath)
+{
+  openfluid::tools::FilesystemPath GitSubmoduleTargetDirectory = openfluid::tools::FilesystemPath(
+        {RootPath.toStdString(), ".git", "modules", ModuleSubPath.toGeneric()});
+  if (GitSubmoduleTargetDirectory.isDirectory())
+  {
+    return GitSubmoduleTargetDirectory.removeDirectory();
+  }
+  return false;
+}
+
+
+// =====================================================================
+// =====================================================================
+
+
+QString GitUIProxy::removeSubmodule(const QString& MainPathString, const QString& SubmodulePathString)
```

Comment:
I need a description of removal process of a git submodule during the code review